adapters/eplanning/eplanning.go

```diff
@@ -61,16 +65,62 @@ type hbResponseAd struct {
 	Height       uint64 `json:"h,omitempty"`
 }
 
+func isPriority(index1 int, index2 int) bool {
+	if index1 > -1 {
+		if index2 > -1 {
+			if index1 < index2 {
+				return false
+			} else {
+				return true
+			}
+		} else {
+			return false
+		}
+	} else {
+		if index2 > -1 {
+			return true
+		} else {
+			return false
+		}
+	}
+}
+
+func lessByPriority(w1, h1, w2, h2 uint64, priorityOrderForSizesAsc []string) bool {
+	size1 := fmt.Sprintf("%dx%d", w1, h1)
+	size2 := fmt.Sprintf("%dx%d", w2, h2)
+	index1 := indexOf(size1, priorityOrderForSizesAsc)
+	index2 := indexOf(size2, priorityOrderForSizesAsc)
+
+	return isPriority(index1, index2)
+}
+
+type byPriorityMobile []openrtb.Format
+
+func (a byPriorityMobile) Len() int      { return len(a) }
+func (a byPriorityMobile) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byPriorityMobile) Less(i, j int) bool {
+	return lessByPriority(a[i].W, a[i].H, a[j].W, a[j].H, priorityOrderForMobileSizesAsc)
+}
```

Comment:
I just want to double-check on the sorting logic. The first element in `priorityOrderForMobileSizesAsc` has the highest priority and should be placed last in the sorted list, right? Just for the sake of understanding the changes better I wrote the test cases shown below. Is the behavior they unveil correct?  Is this the intended logic? (I find test case #4 particularly interesting)
```
// Some test I wrote just to help me understand the sorting logic better
func TestLessByPriorityMobile(t *testing.T) {
	testCases := []struct {
		desc          string
		inFormatList  []openrtb.Format
		expectedOrder []openrtb.Format
	}{
		//priorityOrderForMobileSizesAsc = []string{"1x1", "300x50", "320x50", "300x250"}
		{
			desc: "1) No WxH is listed in priorityOrderForMobileSizesAsc list doesn't change",
			inFormatList: []openrtb.Format{
				{W: 640, H: 480},
				{W: 320, H: 240},
			},
			expectedOrder: []openrtb.Format{
				{W: 640, H: 480},
				{W: 320, H: 240},
			},
		},
		{
			desc: "2) One entry listed in priorityOrderForMobileSizesAsc, that entry will be swapped to the last place",
			inFormatList: []openrtb.Format{
				{W: 1, H: 1},
				{W: 1080, H: 780},
				{W: 640, H: 480},
			},
			expectedOrder: []openrtb.Format{
				{W: 1080, H: 780},
				{W: 640, H: 480},
				{W: 1, H: 1},
			},
		},
		{
			desc: "3) Two entries found in priorityOrderForMobileSizesAsc and will be swapped to appear in order",
			inFormatList: []openrtb.Format{
				{W: 300, H: 250},
				{W: 1, H: 1},
			},
			expectedOrder: []openrtb.Format{
				{W: 300, H: 250},
				{W: 1, H: 1},
			},
		},
		{
			desc: "4) List exactly like priorityOrderForMobileSizesAsc. Entries will be reversed.",
			inFormatList: []openrtb.Format{
				{W: 1, H: 1},
				{W: 300, H: 50},
				{W: 320, H: 50},
				{W: 300, H: 250},
			},
			expectedOrder: []openrtb.Format{
				{W: 300, H: 250},
				{W: 320, H: 50},
				{W: 300, H: 50},
				{W: 1, H: 1},
			},
		},
	}
	for i, test := range testCases {

		actualOrder := test.inFormatList
		sort.Sort(byPriorityMobile(actualOrder))

		assert.Equal(t, test.expectedOrder, actualOrder, "Test %d not ok", i+1)
	}
}
adapters/eplanning/eplanning_test.go
```

The main reason behind wanting to double-check this is because in line 311 shown below it seems to me that the very first non-zero element (that can be the one with lowest priority) would be selected:
```
310         for _, format := range sizesSortedByPriority {
311             if format.W != 0 && format.H != 0 {
312                 return format.W, format.H
313             }
314         }
315     }
316
317     return 0, 0
318 }
adapters/eplanning/eplanning.go
```